mindsdb/libs/phases/stats_generator/stats_generator.py

```diff
@@ -647,6 +648,32 @@ def run(self, input_data, hmd=None, print_logs=True):
                     ,'outlier_score': stats[col_name]['lof_based_outlier_score']
                 }
 
+                # map each bucket to list of outliers in it
+                bucket_outliers = defaultdict(list)
+                for value in stats_v2[col_name]['outliers']['outlier_values']:
+                    vb_index = get_value_bucket(value, stats_v2[col_name]['percentage_buckets'], stats[col_name])
+                    vb = stats_v2[col_name]['percentage_buckets'][vb_index]
+                    bucket_outliers[vb].append(value)
+                
+                stats_v2[col_name]['outliers']['outlier_buckets'] = []
+                for bucket, outlier_values in bucket_outliers.items():
+                    bucket_index = stats_v2[col_name]['histogram']['x'].index(bucket)
+
+                    bucket_values_num = stats_v2[col_name]['histogram']['y'][bucket_index]
+                    bucket_outliers_num = len(outlier_values)
+
+                    total_outliers_num = len(stats_v2[col_name]['outliers']['outlier_values'])
+                    
+                    # Are half of values in the bucket outliers?
+                    predominantly_outlier = bucket_values_num > 0 and (bucket_outliers_num / bucket_values_num) > 0.5
+
+                    # Does number of outliers in the bucket make 95% of outliers
+                    # among all buckets?
+                    percentile_outlier = bucket_outliers_num > 0 and total_outliers_num > 0 and (bucket_outliers_num / total_outliers_num) > (1 - 0.05)
```

Comment:
Hmh, so this is not what we had in mind. What I mean by that was "the number of outliers in this bucket is in the 95th percentile".

Basically meaning that out of e.g. 100 buckets, the 95th percentile are the 5 with the highest number of outliers proportional to the nr of element in the bucket (see np.percentile for example/helper func)

Reply:
So this would have to be computed outside the for once you already know the nr of outliers in each bucket.